_data/data.js

```diff
@@ -191,6 +194,33 @@ const downloadImage = async function (url, file) {
   console.time(label);
   console.log(`Image downloading: ${file}`);
 
+  if (downloadedImagesURLs.has(url)) {
+    // Copy local image instead of downloading again
+    const oldFilePath = downloadedImagesURLs.get(url).path;
+    const extension = oldFilePath.match(/\.[a-z0-9]+$/i);
+    const filename = `${file}${extension}`;
+    const path = Path.join(IMAGE_DIRECTORY, filename);
+    const virtualPath = Path.join(IMAGE_PATH, filename);
+
+    return await FS.copyFile(oldFilePath, path)
+      .then((res) => {
+        console.log(`Image copied: ${path}`);
+
+        sameImageMap.set(path, oldFilePath);
```

Comment:
Speaking of race condition, the use of `sameImageMap` has a similar problem. Let's say A and B are pointing to the same URL and A is downloaded before B starts downloading, we will record B is a copy of A here. However, if we try to start resizing B before A finishes all resizing work (which is async), then copying A's resizing output to B will fail (partially or fully).